diff --git a/src/resources/lib/modules/connman.py b/src/resources/lib/modules/connman.py
index 52bca50..7c2cd14 100644
--- a/src/resources/lib/modules/connman.py
+++ b/src/resources/lib/modules/connman.py
@@ -375,8 +375,9 @@ def dbus_config(self, category):
                 postfix = '.Configuration'
             for entry in sorted(self.struct[category]['settings'], key=lambda x: self.struct[category]['settings'][x]['order']):
                 setting = self.struct[category]['settings'][entry]
-                if (setting['value'] != '' or hasattr(setting, 'changed')) and not 'parent' in setting or 'parent' in setting \
-                    and self.struct[category]['settings'][setting['parent']['entry']]['value'] in setting['parent']['value']:
+                if (setting['value'] != '' or hasattr(setting, 'changed')) \
+                   and (not 'parent' in setting or ('parent' in setting and self.struct[category]['settings'][setting['parent']['entry']]['value'] \
+                                                    in setting['parent']['value'])):
                     if setting['dbus'] == 'Array':
                         value = dbus.Array(dbus.String(setting['value'], variant_level=1).split(','), signature=dbus.Signature('s'),
                                            variant_level=1)

