diff --git a/src/resources/lib/modules/connman.py b/src/resources/lib/modules/connman.py
index 7c2cd14..5d3883c 100644
--- a/src/resources/lib/modules/connman.py
+++ b/src/resources/lib/modules/connman.py
@@ -72,6 +72,7 @@ def __init__(self, servicePath, oeMain):
                                 'value': ['manual'],
                                 },
                             'action': 'set_value',
+                            'notempty': True,
                             },
                         'Netmask': {
                             'order': 3,
@@ -84,6 +85,7 @@ def __init__(self, servicePath, oeMain):
                                 'value': ['manual'],
                                 },
                             'action': 'set_value',
+                            'notempty': True,
                             },
                         'Gateway': {
                             'order': 4,
@@ -96,6 +98,7 @@ def __init__(self, servicePath, oeMain):
                                 'value': ['manual'],
                                 },
                             'action': 'set_value',
+                            'notempty': True,
                             },
                         },
                     },
@@ -375,7 +378,7 @@ def dbus_config(self, category):
                 postfix = '.Configuration'
             for entry in sorted(self.struct[category]['settings'], key=lambda x: self.struct[category]['settings'][x]['order']):
                 setting = self.struct[category]['settings'][entry]
-                if (setting['value'] != '' or hasattr(setting, 'changed')) \
+                if (setting['value'] != '' or (hasattr(setting, 'changed') and not hasattr(setting, 'notempty'))) \
                    and (not 'parent' in setting or ('parent' in setting and self.struct[category]['settings'][setting['parent']['entry']]['value'] \
                                                     in setting['parent']['value'])):
                     if setting['dbus'] == 'Array':

