diff --git a/src/resources/lib/oeWindows.py b/src/resources/lib/oeWindows.py
index 38c9f52..1b1a1c2 100644
--- a/src/resources/lib/oeWindows.py
+++ b/src/resources/lib/oeWindows.py
@@ -298,6 +298,8 @@ def onClick(self, controlID):
                     if returnValue != '' and returnValue != '/':
                         selectedItem.setProperty('value', str(returnValue))
                 elif strTyp == 'ip':
+                    if strValue == '':
+                        strValue = '0.0.0.0'
                     xbmcDialog = xbmcgui.Dialog()
                     returnValue = xbmcDialog.numeric(3, 'LibreELEC.tv', strValue)
                     if returnValue != '':

